### DISCUSSION TITLE
Have stream deck return to 1st page of profile after a given period of time

### DISCUSSION THREAD

**Discussion by roveer24** [elgatosf/elgato#153] [Help] [answered]
I'm in the process of creating a multi-page profile for my stream deck neo.  Is there a way to have the neo automatically return to the 1st page after a given period of time, say 30 seconds?  

Many thanks.

**Comment by DhSufi** [comment, marked as answer]
Changing pages is only possible if the profile is distributed with your plugin. You can learn how to build a profile for your pllugin here: https://docs.elgato.com/streamdeck/sdk/guides/profiles

Once you are in your own profile. you can change to pages using the switchToProfile(device, profileName, Page) function.
Also note if you do not pass a profile name, then it siwtches automatically to the previous profile.

Finally to set a period of time you can use setTimeout(). However be careful with it, as long as you need to handle the cancelation of the timeout when necessary, otherwise it will run the callback.
